### PR TITLE
Handle case where there are no replicas in ideal state

### PR DESCRIPTION
**Pull request by rven1**

### CODE REVIEW

File: pinot-controller/src/test/java/com/linkedin/pinot/controller/helix/SegmentStatusCheckerTest.java [hunk 1]
```diff
@@ -367,4 +367,47 @@ public void missingEVPartitionPushTest() throws Exception {
     segmentStatusChecker.stop();
   }
 
+  @Test
+  public void noReplicas() throws Exception {
```
Comment (jfim) [inline pinot-controller/src/test/java/com/linkedin/pinot/controller/helix/SegmentStatusCheckerTest.java]: Rename test to `testZeroReplicas`

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java [hunk 2]
```diff
@@ -242,8 +242,10 @@ public void runSegmentMetrics() {
       // Synchronization provided by Controller Gauge to make sure that only one thread updates the gauge
       _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.NUMBER_OF_REPLICAS,
           nReplicasExternal);
-      _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.PERCENT_OF_REPLICAS,
-          nReplicasExternal*100/nReplicasIdealMax);
+      if (nReplicasIdealMax != 0) {
+        _metricsRegistry.setValueOfTableGauge(tableName, ControllerGauge.PERCENT_OF_REPLICAS,
```
Comment (antumbde) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java]: We won't emit a value for nReplicasIdealMax = 0. Seems there should be one. What happens if we've alerts on this and no value is emitted for a while ?

Comment (rven1) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java]: We have set it to 0 in setStatusToDefault

Comment (antumbde) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java]: that happens only on startup (or for non-leaders). what about on a continuous basis for leader ?

**Pull request merged**